### ISSUE TITLE
feat: Add azure_keyvault_key_rotation_policy to Azure source plugin

### ISSUE THREAD

**Issue by code-vise** [cloudquery/cloudquery#14292]
### Describe the resource

Need data from this api https://learn.microsoft.com/en-us/rest/api/keyvault/keys/get-key-rotation-policy/get-key-rotation-policy?tabs=HTTP

### Use Case

Would like to check weather the keys in key vaults have rotation policy enabled or not, and what is the rotation time.

### Link to API documentation

https://learn.microsoft.com/en-us/rest/api/keyvault/keys/get-key-rotation-policy/get-key-rotation-policy?tabs=HTTP

### Pull request (optional)

- [ ] I can submit a pull request

**Comment by bbernays** [comment]
@john-s58 - I think you had looked into this and found that the data is available in another table, Can you document your findings here?

**Comment by john-s58** [comment]
This data should be available in the [keyvault_keys table](https://hub.cloudquery.io/plugins/source/cloudquery/azure/latest/tables/azure_keyvault_keyvault_keys), specifically as a property within the JSON column Properties->RotationPolicy. But, upon testing it appears that some properties are not being retrieved as expected. These are all the properties: https://pkg.go.dev/github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/keyvault/armkeyvault#KeyProperties and I only saw 2 of those in the resources I created. We may need to raise an issue with the azure-sdk-for-go repository. I will do more tests to ensure that nothing is being overlooked before opening an issue.
